autora/experimentalist/sampler/dissimilarity.py

```diff
@@ -0,0 +1,103 @@
+from typing import Iterable
+
+import numpy as np
+from sklearn.metrics import DistanceMetric
+
+
+def summed_dissimilarity_sampler(
+    X: np.ndarray, X_ref: np.ndarray, n: int = 1, metric: str = "euclidean"
+) -> np.ndarray:
+    """
+    This dissimilarity samples re-arranges the pool of IV conditions according to their
+    dissimilarity with respect to a reference pool X_base. The default dissimilarity is calculated
+    as the average of the pairwise distances between the conditions in X and X_base.
+
+    Args:
+        X: pool of IV conditions to evaluate dissimilarity
+        X_ref: reference pool of IV conditions
+        n: number of samples to select
+        metric: dissimilarity measure. Options: 'euclidean', 'manhattan', 'chebyshev',
+            'minkowski', 'wminkowski', 'seuclidean', 'mahalanobis', 'haversine',
+            'hamming', 'canberra', 'braycurtis', 'matching', 'jaccard', 'dice',
+            'kulsinski', 'rogerstanimoto', 'russellrao', 'sokalmichener',
+            'sokalsneath', 'yule'. See `sklearn.metrics.DistanceMetric` for more details.
+
+    Returns:
+        Sampled pool
+    """
+
+    if isinstance(X, Iterable):
+        X = np.array(list(X))
+
+    if isinstance(X_ref, Iterable):
+        X_ref = np.array(list(X_ref))
+
+    if X.ndim == 1:
+        X = X.reshape(-1, 1)
+
+    if X_ref.ndim == 1:
+        X_ref = X_ref.reshape(-1, 1)
+
+    if X.shape[1] != X_ref.shape[1]:
+        raise ValueError(
+            f"X and X_ref must have the same number of columns.\n"
+            f"X has {X.shape[1]} columns, while X_ref has {X_ref.shape[1]} columns."
+        )
+
+    if X.shape[0] < n:
+        raise ValueError(
+            f"X must have at least {n} rows matching the number of requested samples."
+        )
+
+    valid_metrics = [
+        "euclidean",
+        "manhattan",
+        "chebyshev",
+        "minkowski",
+        "wminkowski",
+        "seuclidean",
+        "mahalanobis",
+        "haversine",
+        "hamming",
+        "canberra",
+        "braycurtis",
+        "matching",
+        "dice",
+        "kulsinski",
+        "rogerstanimoto",
+        "russellrao",
+        "sokalmichener",
+        "sokalsneath",
+        "yule",
+    ]
+
+    if metric not in valid_metrics:
+        raise ValueError(
+            f"Unsupported metric: '{metric}'\n" f"Only {valid_metrics} is supported."
+        )
+    else:
+        dist = DistanceMetric.get_metric(metric)
```

Comment:
can we get away with just doing a one liner here:
```suggestion
    dist = DistanceMetric.get_metric(metric)
```
Will that fail sufficiently loudly for us? 

Combined with the type hint in the function signature and the MyPy checker, that will mean that most people won't use it wrong, but a sufficiently clever person will be able to implement their own metric, if they so choose. (They can add a new metric to scikit-learn and automatically be able to import it here.)